packages/arb-token-bridge-ui/src/hooks/useOftTransactionHistory.ts

```diff
@@ -297,9 +307,14 @@ export function useOftTransactionHistory({
 
     const layerZeroResponse: LayerZeroResponse = await response.json()
 
-    return layerZeroResponse.data
-      .filter(validateLayerZeroMessage) // filter out transactions that don't have Arbitrum supported chain ids, and USDT0 protocol
-      .map(mapLayerZeroMessageToLayerZeroTransaction)
+    const validMessages = []
+    for (const message of layerZeroResponse.data) {
+      if (await validateLayerZeroMessage(message)) {
+        validMessages.push(message)
+      }
+    }
+
+    return validMessages.map(mapLayerZeroMessageToLayerZeroTransaction)
```

Comment:
i would use `Promise.all` here so we can run them in parallel

Reply:
If we're scared of RPC rate limit, we could maybe batch them in chunk

Reply:
Discussed offline, we do have RPC calls as part of this loop, and we will batch all RPC code in a different PR